server/topic.go

```diff
@@ -338,22 +381,28 @@ func (t *Topic) runLocal(hub *Hub) {
 
 			if t.isInactive() {
 				if !asUid.IsZero() && leave.id != "" {
-					leave.sess.queueOut(ErrLocked(leave.id, t.original(asUid), now))
+					leave.sess.queueOutWithOverrides(ErrLocked(leave.id, t.original(asUid), now), leave.sessOverrides)
 				}
 				continue
 
 			} else if leave.unsub {
 				// User wants to leave and unsubscribe.
 				// asUid must not be Zero.
-				if err := t.replyLeaveUnsub(hub, leave.sess, asUid, leave.id, nil/*leave.sessOverrides*/); err != nil {
+				if err := t.replyLeaveUnsub(hub, leave.sess, asUid, leave.id, leave.sessOverrides); err != nil {
 					log.Println("failed to unsub", err, leave.sess.sid)
 					continue
 				}
 
-			} else if pssd := t.remSession(leave.sess, asUid); pssd != nil {
+			} else if pssd := t.maybeRemoveSession(leave.sess, asUid, /*doRemove=*/!leave.sess.isProxy || leave.terminateProxyConnection); pssd != nil {
 				// Just leaving the topic without unsubscribing if user is subscribed.
 
-				pud := t.perUser[pssd.uid]
+				var uid types.Uid
+				if t.isProxy {
```

Comment:
This section is probably incorrect. 

* Behavior at the [proxy topic + normal session] should match that of the [normal topic + normal session]:
`pud := t.perUser[pssd.uid]` like before.
* Behavior of [normal topic + proxy session] should be different: it should not care abut updating any of `t.perUser[uid] = pud` because proxy session multiplexes many uses. 

See comment at declaration of `maybeRemoveSession()`.

Reply:
OK